### DISCUSSION TITLE
Values in field "data" in table twofactor are in plain text

### DISCUSSION THREAD

**Discussion by MButcho** [dani-garcia/vaultwarden#2882] [General]
When using SQL command "select * from twofactor", the 2FA is in plain text. It will be good to encrypt it via master password.

**Comment by BlackDex** [comment]
@MButcho, that isn't really useful, since we still need it decrypted to verify if the TOTP is correct.
Also, we can't encrypt it with the master-password, since we do not have the master-password.

We only receive a hash which we can verify if the password is correct or not, and using that value to en/decrypt it is useless, since that is stored in the database also, and thus can be used to decrypt it.
